### DISCUSSION TITLE
Did the kernel starting mechanism change in a recent pre-release?

### DISCUSSION THREAD

**Discussion by leifwalsh** [microsoft/vscode-jupyter#13106] [Questions and Answers]
I run the pre-release version of this extension to try to stay ahead of regressions, and I think I noticed one in our environment. I think this might be related to the "streamlining" mentioned in https://github.com/microsoft/vscode-jupyter/discussions/12999. 

Is it the case that, in the non-remote jupyter server mode, the extension now runs `jupyter notebook` to start a local notebook and then connects to that to start kernels? Is it the case that it used to directly run the kernel and interact with it as if the extension were the jupyter server?

If so, I think something we've done in our environment makes jupyter in vscode break with this change and I need to figure out what we can do about it before this change hits normal releases.

The thing we've done is that we have a customized version of jupyter internally, and to avoid confusion the team that owns it changed the `jupyter` and `jupyter-notebook` commands to print an error (instructing the user to use the custom command name) and exit. I think this makes everything break with the pre-release version of vscode-jupyter.

I realize this is a very https://xkcd.com/1172/ style "bug", so I'm starting this as a discussion just to find out if that is what happened first.

**Comment by DonJayamanne** [comment]
@leifwalsh please can you file a bug for this issue 
It could be a regression, I will look into this asap

and to answer your question yes, yes we did change a few things
I am assuming you are using a Linux is
Please do file a bug with the following logs

Please could you enable logging as follows:
* Open settings & go into `Jupyter->Logging`
* Change the value to `verbose`
* Reload VS Code,
* Attempt to repro the issue & then copy the output from the `Jupyter` output panel.

**Comment by DonJayamanne** [comment]
@leifwalsh  please do create an issue when possible
As you seem to be running into a regression that has the potential to ship in the next release and that’s something I would like to avoid

**Comment by leifwalsh** [comment]
Thanks. Did my best here: https://github.com/microsoft/vscode-jupyter/issues/13107 Let's continue there and if you have other questions I can try to answer them.
